src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/file/FileStorage.scala

```diff
@@ -0,0 +1,92 @@
+package ch.epfl.bluebrain.nexus.kg.resources.file
+import java.nio.file.{Files, Path, Paths}
+import java.time.Instant
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.stream.scaladsl.{FileIO, Keep}
+import akka.stream.{ActorMaterializer, Materializer}
+import cats.effect.{Effect, IO}
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.kg.KgError
+import ch.epfl.bluebrain.nexus.kg.resources.file.File.{Digest, FileAttributes, FileDescription, StoredSummary}
+import ch.epfl.bluebrain.nexus.kg.resources.{ProjectRef, ResId}
+import ch.epfl.bluebrain.nexus.rdf.Iri.AbsoluteIri
+import journal.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.control.NonFatal
+
+case class FileStorage(ref: ProjectRef,
+                       id: AbsoluteIri,
+                       uuid: UUID,
+                       rev: Long,
+                       instant: Instant,
+                       deprecated: Boolean,
+                       default: Boolean,
+                       digestAlgorithm: String,
+                       volume: Path)
+    extends Storage
+
+object FileStorage {
+
+  private val logger: Logger = Logger[this.type]
+
+  /**
+    * Wraps both the absolute and relative information about the file location
+    *
+    * @param path     absolute Path where to find the file
+    * @param relative relative route to the file location
+    */
+  final case class Location(path: Path, relative: Path)
```

Comment:
we can make this private, since it is scoped on this method. And we can remove the docs then